frontend/app/src/pages/balances/blockchain/index.vue

```diff
@@ -67,47 +55,16 @@ const { ksmAccounts, dotAccounts, avaxAccounts, optimismAccounts } =
   useChainAccountBalances();
 const { btcAccounts, bchAccounts } = useBtcAccountBalances();
 
-const blockchainData: BlockchainData = {
-  btcAccounts,
-  bchAccounts,
-  dotAccounts,
-  ethAccounts,
-  eth2Accounts,
-  avaxAccounts,
-  ksmAccounts,
-  loopringAccounts,
-  optimismAccounts
-};
-
 const { blockchainAssets } = useBlockchainAggregatedBalances();
 
-const getFirstContext = (data: BlockchainData) => {
-  const hasData = (data: Ref<BlockchainAccountWithBalance[]>) =>
-    get(data).length > 0;
-
-  if (hasData(data.btcAccounts)) {
-    return Blockchain.BTC;
-  } else if (hasData(data.bchAccounts)) {
-    return Blockchain.BCH;
-  } else if (hasData(data.ksmAccounts)) {
-    return Blockchain.KSM;
-  } else if (hasData(data.dotAccounts)) {
-    return Blockchain.DOT;
-  } else if (hasData(data.avaxAccounts)) {
-    return Blockchain.AVAX;
-  } else if (hasData(data.optimismAccounts)) {
-    return Blockchain.OPTIMISM;
-  }
-
-  return Blockchain.ETH;
-};
-
 const context: ComputedRef<Blockchain> = computed(() => {
   const intersect = get(intersections);
-  let currentContext = getFirstContext(blockchainData);
+  let currentContext = Blockchain.ETH;
+  // pick only intersections that are visible (at least 50%)
+  const activeObservers = pickBy(intersect, e => e);
 
-  for (const current in Blockchain) {
-    if (intersect[current as Blockchain]) {
+  for (const current in activeObservers) {
+    if (activeObservers[current]) {
       currentContext = current as Blockchain;
     }
   }
```

Comment:
Since the `activeObservers` are filtered with value=true above, you don't need this if anymore, don't you.
Also in this case you know the keys of `activeObservers`, but you will only used the last key of it, isn't it?

If that's the case, will this code become more efficient?
```
const context: ComputedRef<Blockchain> = computed(() => {
  const intersect = get(intersections);
  // pick only intersections that are visible (at least 50%)
  const activeObservers = Object.keys(pickBy(intersect, e => e));
  if (activeObservers.length > 0) {
    return activeObservers[activeObservers.length - 1] as Blockchain;
  }

  return Blockchain.ETH;
});
```